### ISSUE TITLE
`everware-cli`, what should it do?

### ISSUE THREAD

**Issue by betatim** [everware/everware#36]
`everware` is based on the idea that using a docker container during development of an analysis benefits the researcher. This means they keep it working/running for their own benefit. The second part,  a waste product if you wish, is that we get re-usable data analysis. So far everware only really addresses the second part.

To make the first part true we need some tools to help people create docker containers for their analysis and something better/more general than [`run_local.sh`](https://github.com/everware/container-tools/blob/master/run_local.sh).

Enter `everware-cli` a set of command line tools for the "developing your analysis" part. The big question is: what should these tools do?

Features:
- `everware init` create a good default `Dockerfile` in PWD
- `everware build` build the `Dockerfile` into an image
- `everware run` run container as it would on everware webpage, mounting PWD instead of `git clone`'ing the code
- `everware exec` run a command in the running container

//cc @alexpearce who had some ideas too

**Comment by lukasheinrich** [comment]
it would be great if this would somehow natural support developing a container in multiple sessions. i.e under the hood some cycle of docker pull/attach/save/tag/push which might be a bit smarter than what docker provides natively.. (think virtualenv + pip install -e .)

**Comment by betatim** [comment]
What is the workflow here? Asking because the only way I create docker images is to edit `Dockerfile`, build, run, test, edit `Dockerfile`, ...

In the end for `everware` we need to produce a `Dockerfile` that creates the image that is needed. I could imagine a `everware save` that suspends the current container and let's you resume it later on though. Is this what you had in mind?

**Comment by lukasheinrich** [comment]
yeah something like give back the user an session ID (could be straight up the container ID, or for security something else) and allow the user to attach/detach/save. What's the lifetime of the container currently?

**Comment by betatim** [comment]
Look at http://yeoman.io/ to get started on how to setup a "project generator", this would be for the `everware init` part
